From 2ba2ae50587347e8fd4d530196a447a861281f07 Mon Sep 17 00:00:00 2001
From: Hongmei Gou <h-gou@ti.com>
Date: Wed, 25 Mar 2015 11:37:54 -0400
Subject: [PATCH] ARM: DRA7: dts: Add coprocessor nodes for iva.

Add coprocessor nodes for IVAHD.
Set initial OPP to OPP_HIGH.

This patch is recreated from omap kernel tree from commmit ID 8c9f3bbbf9ffc294ca0e19a90963c295f55452d9
---
 arch/arm/boot/dts/dra72x.dtsi |   28 ++++++++++++++++++++++++++++
 arch/arm/boot/dts/dra74x.dtsi |   28 ++++++++++++++++++++++++++++
 2 files changed, 56 insertions(+)

diff --git a/arch/arm/boot/dts/dra72x.dtsi b/arch/arm/boot/dts/dra72x.dtsi
index 2620bc4..4db7178 100644
--- a/arch/arm/boot/dts/dra72x.dtsi
+++ b/arch/arm/boot/dts/dra72x.dtsi
@@ -38,6 +38,34 @@
 		};
 	};
 
+        iva_coproc {
+                compatible = "ti,coproc";
+                clocks = <&dpll_iva_m2_ck>, <&dpll_iva_ck>;
+                clock-names = "fclk", "dpll";
+                clock-target-frequency = <532000000>;
+                operating-points = <
+                        388200  1055000
+                        430000  1150000
+                        532000  1250000
+                        >;
+                coproc-voltdm = <&voltdm_ivahd>;
+                voltage-tolerance = <1>;
+        };
+
+        dsp_coproc {
+                compatible = "ti,coproc";
+                clocks = <&dpll_dsp_m2_ck>, <&dpll_dsp_ck>;
+                clock-names = "fclk", "dpll";
+                clock-target-frequency = <700000000>;
+                operating-points = <
+                        600000  1055000
+                        700000  1150000
+                        700000  1250000
+                        >;
+                coproc-voltdm = <&voltdm_dspeve>;
+                voltage-tolerance = <1>;
+        };
+
 	thermal_zones: thermal-zones {
 		#include "omap5-cpu-thermal.dtsi"
 	};
diff --git a/arch/arm/boot/dts/dra74x.dtsi b/arch/arm/boot/dts/dra74x.dtsi
index c6ae214..e668fc1 100644
--- a/arch/arm/boot/dts/dra74x.dtsi
+++ b/arch/arm/boot/dts/dra74x.dtsi
@@ -43,6 +43,34 @@
 		};
 	};
 
+        iva_coproc {
+                compatible = "ti,coproc";
+                clocks = <&dpll_iva_m2_ck>, <&dpll_iva_ck>;
+                clock-names = "fclk", "dpll";
+                clock-target-frequency = <532000000>;
+                operating-points = <
+                        388200  1055000
+                        430000  1150000
+                        532000  1250000
+                        >;
+                coproc-voltdm = <&voltdm_ivahd>;
+                voltage-tolerance = <1>;
+        };
+
+        dsp_coproc {
+                compatible = "ti,coproc";
+                clocks = <&dpll_dsp_m2_ck>, <&dpll_dsp_ck>;
+                clock-names = "fclk", "dpll";
+                clock-target-frequency = <700000000>;
+                operating-points = <
+                        600000  1055000
+                        700000  1150000
+                        700000  1250000
+                        >;
+                coproc-voltdm = <&voltdm_dspeve>;
+                voltage-tolerance = <1>;
+        };
+
 	thermal_zones: thermal-zones {
 		#include "omap5-cpu-thermal.dtsi"
 	};
-- 
1.7.9.5

